### PR TITLE
Adcm-1845_1 add tests to provider page

### PR DESCRIPTION
**Pull request by annademina**

### CODE REVIEW

File: tests/ui_tests/test_provider_page.py [hunk 1]
```diff
@@ -147,3 +196,125 @@ def test_run_upgrade_on_provider_page_by_toolbar(self, app_fs, sdk_client_fs, up
...
+        provider_action_page.check_all_elements()
+
+    @pytest.mark.smoke()
+    def test_run_action_on_provider_action_page(self, app_fs, upload_and_create_test_provider):
+        provider_action_page = ProviderActionsPage(
+            app_fs.driver, app_fs.adcm.url, upload_and_create_test_provider.id
+        ).open()
+        provider_action = provider_action_page.get_all_actions()[0]
+        provider_action_page.click_run_btn_in_action(provider_action)
+        provider_action_page.check_empty_page()
```
Comment (Sealwing) [inline tests/ui_tests/test_provider_page.py:316]: `check_empty_page` is confusing here. I suppose it means that no actions are presented, while one of them is running. So maybe call it `check_no_actions_presented` or smt like it?

File: tests/ui_tests/test_provider_page.py [hunk 2]
```diff
@@ -147,3 +196,125 @@ def test_run_upgrade_on_provider_page_by_toolbar(self, app_fs, sdk_client_fs, up
...
+        config_row = provider_config_page.config.get_all_config_rows()[0]
+        provider_config_page.config.type_in_config_field(params['wrong_value'], row=config_row)
+        provider_config_page.config.check_field_is_invalid(params['not_req_name'])
+
+
+class TestProviderActionPage:
+    @pytest.mark.smoke()
+    def test_open_by_tab_provider_action_page(self, app_fs, upload_and_create_test_provider):
+        provider_main_page = ProviderMainPage(app_fs.driver, app_fs.adcm.url, upload_and_create_test_provider.id).open()
+        provider_main_page.open_actions_tab()
```
Comment (Sealwing) [inline tests/ui_tests/test_provider_page.py:304]: Maybe make `open_actions_tab` returning `ProviderActionsPage`? For detailed pages, it looks like nice practice to add in our codebase

File: tests/ui_tests/test_provider_page.py [hunk 3]
```diff
@@ -147,3 +196,125 @@ def test_run_upgrade_on_provider_page_by_toolbar(self, app_fs, sdk_client_fs, up
...
+            "row_value_new": "test",
+            "row_value_old": "0000",
+            "config_name": "test_name",
+        }
+        provider_config_page = ProviderConfigPage(
+            app_fs.driver, app_fs.adcm.url, upload_and_create_test_provider.id
+        ).open()
+        config_row = provider_config_page.config.get_all_config_rows()[0]
+        provider_config_page.config.type_in_config_field(
+            row=config_row, value=params["row_value_new"], clear=True, adcm_test=None
```
Comment (Sealwing) [inline tests/ui_tests/test_provider_page.py:272]: `adcm_test` is no longer an argument, this call will fall

File: tests/ui_tests/test_provider_page.py [hunk 4]
```diff
@@ -147,3 +196,125 @@ def test_run_upgrade_on_provider_page_by_toolbar(self, app_fs, sdk_client_fs, up
...
+            "row_value_old": "0000",
+            "config_name_new": "test_name",
+            "config_name_old": "init",
+        }
+        provider_config_page = ProviderConfigPage(
+            app_fs.driver, app_fs.adcm.url, upload_and_create_test_provider.id
+        ).open()
+        config_row = provider_config_page.config.get_all_config_rows()[0]
+        provider_config_page.config.type_in_config_field(
+            row=config_row, value=params["row_value_new"], clear=True, adcm_test=None
```
Comment (Sealwing) [inline tests/ui_tests/test_provider_page.py:250]: Remove `adcm_test`, see comment below

File: tests/ui_tests/app/page/provider_list/page.py [hunk 5]
```diff
@@ -96,3 +103,10 @@ def wait_state():
             assert state_after != self.table.LOADING_STATE_TEXT
 
         wait_until_step_succeeds(wait_state, period=1, timeout=self.default_loc_timeout)
+
+    @allure.step("Delete host")
+    def delete_provider_by_row(self, row: WebElement):
```
Comment (Sealwing) [inline tests/ui_tests/app/page/provider_list/page.py:108]: Above you use `click_name_in_row`, here you use **by**. I think it's better to use "in"

**Review by Sealwing — CHANGES_REQUESTED** [review]

File: tests/ui_tests/test_provider_page.py (hunk 1 above)
Comment (annademina) [reply tests/ui_tests/test_provider_page.py]: renamed

File: tests/ui_tests/test_provider_page.py (hunk 2 above)
Comment (annademina) [reply tests/ui_tests/test_provider_page.py]: changed

File: tests/ui_tests/test_provider_page.py (hunk 3 above)
Comment (annademina) [reply tests/ui_tests/test_provider_page.py]: removed

File: tests/ui_tests/test_provider_page.py (hunk 4 above)
Comment (annademina) [reply tests/ui_tests/test_provider_page.py]: removed

File: tests/ui_tests/app/page/provider_list/page.py (hunk 5 above)
Comment (annademina) [reply tests/ui_tests/app/page/provider_list/page.py]: renamed

File: tests/ui_tests/app/page/provider/page.py [hunk 6]
```diff
@@ -38,22 +50,78 @@ class ProviderPageMixin(BasePageObject):
...
+
+    @allure.step("Run action")
+    def click_run_btn_in_action(self, action: WebElement):
+        self.find_child(action, CommonActionLocators.ActionCard.play_btn).click()
+        self.wait_element_visible(ActionDialog.body)
+        self.find_and_click(ActionDialog.run)
+
+    @allure.step("Check that action page is empty")
+    def check_no_actions_presented(self):
+        assert "Nothing to display." in self.find_element(CommonActionLocators.info_text).text
```
[the remark this answers is not in the document]
Comment (annademina) [reply tests/ui_tests/app/page/provider/page.py]: added

File: tests/ui_tests/app/page/provider/page.py [hunk 7]
```diff
@@ -38,22 +50,78 @@ class ProviderPageMixin(BasePageObject):
...
+
+    MENU_SUFFIX = 'action'
+
+    MAIN_ELEMENTS = [
+        ObjectPageLocators.title,
+        ObjectPageLocators.subtitle,
+        CommonActionLocators.action_card,
+    ]
+
+    def get_all_actions(self):
```
[the remark this answers is not in the document]
Comment (annademina) [reply tests/ui_tests/app/page/provider/page.py]: added

File: tests/ui_tests/app/page/provider/page.py [hunk 8]
```diff
@@ -38,22 +50,78 @@ class ProviderPageMixin(BasePageObject):
...
         self.footer = PageFooter(self.driver, self.base_url)
         self.config = CommonConfigMenuObj(self.driver, self.base_url)
-        self.cluster_id = provider_id
+        self.provider_id = provider_id
         self.toolbar = CommonToolbar(self.driver, self.base_url)
         self.table = CommonTableObj(self.driver, self.base_url)
 
+    def open_main_tab(self):
+        self.find_and_click(ObjectPageMenuLocators.main_tab)
+        return ProviderMainPage(self.driver, self.base_url, self.provider_id)
```
[the remark this answers is not in the document]
Comment (annademina) [reply tests/ui_tests/app/page/provider/page.py]: added

[the remark this answers is not in the document]
Comment (annademina) [reply tests/ui_tests/app/page/provider/page.py]: added

**Review by Sealwing — APPROVED** [review]

**Review by skhomuti — APPROVED** [review]

**Pull request merged**